staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -150,6 +151,43 @@ var (
 		[]string{"requestKind"},
 	)
 
+	// low and high concurrency watermarks from the last few windows
+	concurrencyWatermarks = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Name:           "apiserver_inflight_count_watermarks",
+			Help:           "Low and high watermark of mutating, readonly requests executing in recent windows.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"requestKind", "mark", "lag"},
```

Comment:
what is the cardinality of lag and mark?

Reply:
There are two values of "mark": "low" and "high".  There are 14 values of "lag": "1" through "14".

Reply:
At present the window width and number of windows are constants.  At the cost of a configuration API change, we could make either of those variable.